dds_web/api/errors.py

```diff
@@ -35,55 +36,16 @@ class ItemDeletionError(exceptions.HTTPException):
 ####################################################################################################
 
 
-class IncorrectDecoratorUsageException(exceptions.HTTPException):
-    """Errors occuring in DDS decorators, e.g. due to incorrect order or overall usage"""
-
-    def __init__(self, message):
-        super().__init__(message)
-
-        general_logger.warning(message)
-
-
-# ##################################################################################################
-
-
 class AuthenticationError(exceptions.HTTPException):
     """Base class for errors due to authentication failure."""
 
-    def __init__(self, message, username=None, project=None):
+    def __init__(self, message="Missing or incorrect credentials"):
         super().__init__(message)
 
         general_logger.warning(message)
 
-        action_logger.warning(
-            message,
-            extra={
-                **extra_info,
-                "current_user": username,
-                "action": actions.get(flask.request.endpoint),
-                "project": project,
-            },
-        )
-
-
-class InvalidUserCredentialsError(exceptions.HTTPException):
-    """Errors occurring during user authentication."""
-
-    def __init__(self, username, message="Incorrect username and/or password."):
-        super().__init__(message)
-
-        action_logger.warning(
-            message,
-            extra={
-                **extra_info,
-                "current_user": username,
-                "action": actions.get(flask.request.endpoint),
-                "project": None,
-            },
-        )
 
-
-class ProjectPermissionsError(exceptions.HTTPException):
+class AccessDeniedError(exceptions.HTTPException):
```

Comment:
@worukan I think we should add the ID to the user table again. We need to be able to log which username attempts a specific action and since the username is the primary key at the moment this generates security warnings.

Reply:
Fixed! I think there is no need to bring back the previously removed ID.